### ISSUE TITLE
We are getting extra character empty classes

### ISSUE THREAD

**Issue by bksubhuti** [daohoangson/flutter_widget_from_html#963]
I cannot build a small app to show you this.. but please run this html code in your widget and see what you get.

You should find an extra no print character /ufffc

https://github.com/flutter/flutter/issues/111931

The first word is **not selectable** because we get this same /ufffc character.  
The issue seems to be with the HTML generation .. There was also an issue posted related to this elsewhere higher up on the framework.

The issue is close, but maybe you have more power if it is in your code.
I'm not smart enough to pull out all of the stuff and then get the output from the htmlwidget.  But my programmer found it and this happens at the beginning of every page when a paragraph marker starts.

```
<p class="bodytext"><a name="para7"></a><a name="para7_mn1"></a><span class="paranum">7</span>. yopi <a name="M1.0005"></a> so, bhikkhave, bhikkhu sekkho <span class="note">[sekho (sī. syā. kaṃ. pī.)]</span> appattamānaso anuttaraṃ yogakkhemaṃ patthayamāno viharati, sopi pathaviṃ pathavito abhijānāti, pathaviṃ pathavito abhiññāya <span class="note">[abhiññatvā (ka.)]</span> pathaviṃ mā maññi <span class="note">[vā maññati]</span>, pathaviyā mā maññi, pathavito mā maññi, pathaviṃ meti mā maññi, pathaviṃ mābhinandi <span class="note">[vā abhinandati (sī.) ṭīkā oloketabbā]</span>. taṃ kissa hetu? ‘pariññeyyaṃ tassā’ti vadāmi.</p>
<p class="bodytext">āpaṃ…pe… tejaṃ… vāyaṃ… bhūte… deve… pajāpatiṃ… brahmaṃ… ābhassare… subhakiṇhe… vehapphale… abhibhuṃ… ākāsānañcāyatanaṃ… viññāṇañcāyatanaṃ… ākiñcaññāyatanaṃ… nevasaññānāsaññāyatanaṃ… diṭṭhaṃ… sutaṃ… mutaṃ… viññātaṃ… ekattaṃ… nānattaṃ… sabbaṃ… nibbānaṃ nibbānato abhijānāti, nibbānaṃ nibbānato abhiññāya <a name="V1.0006"></a> nibbānaṃ mā maññi, nibbānasmiṃ mā maññi, nibbānato mā maññi, nibbānaṃ meti mā maññi, nibbānaṃ mābhinandi. taṃ kissa hetu? ‘pariññeyyaṃ tassā’ti vadāmi.</p>
<p class="centered">sekkhavasena <span class="note">[satthāravasena (sī.), satthuvasena (syā. ka.)]</span> dutiyanayabhūmiparicchedo niṭṭhito.</p>
<p class="bodytext"><a name="para8"></a><a name="para8_mn1"></a><span class="paranum">8</span>. yopi so, bhikkhave, bhikkhu arahaṃ khīṇāsavo vusitavā katakaraṇīyo ohitabhāro anuppattasadattho parikkhīṇabhavasaṃyojano sammadaññā vimutto, sopi pathaviṃ pathavito abhijānāti, pathaviṃ pathavito <a name="T1.0007"></a> abhiññāya pathaviṃ na maññati, pathaviyā na maññati, pathavito na maññati, pathaviṃ meti na maññati, pathaviṃ nābhinandati. taṃ kissa hetu? ‘pariññātaṃ tassā’ti vadāmi.</p>
<p class="bodytext">āpaṃ…pe… tejaṃ… vāyaṃ… bhūte… deve… pajāpatiṃ… brahmaṃ… ābhassare… subhakiṇhe… vehapphale… abhibhuṃ… ākāsānañcāyatanaṃ… viññāṇañcāyatanaṃ… ākiñcaññāyatanaṃ… nevasaññānāsaññāyatanaṃ… diṭṭhaṃ… sutaṃ… mutaṃ… viññātaṃ… ekattaṃ… nānattaṃ… sabbaṃ… nibbānaṃ nibbānato abhijānāti, nibbānaṃ nibbānato abhiññāya nibbānaṃ na maññati, nibbānasmiṃ na maññati, nibbānato na maññati, nibbānaṃ meti na maññati, nibbānaṃ nābhinandati. taṃ kissa hetu? ‘pariññātaṃ tassā’ti vadāmi.</p>
<p class="centered">khīṇāsavavasena tatiyanayabhūmiparicchedo niṭṭhito.</p>

```

**Comment by daohoangson** [comment]
It looks like the selection cannot go backward from `so, bhikkhave`, I could not select `7. yopi` this way. Is this the bug you are describing?

<img width="912" alt="Screen Shot 2023-07-12 at 10 01 53" src="https://github.com/daohoangson/flutter_widget_from_html/assets/239336/974f48c8-27b4-49d7-87ea-23d58e64e9d1">

I think this empty A tag is the culprit `<a name="M1.0005"></a>`. This package renders an empty element as the anchor for this tag. For some reason the `SizedBox` interferes with text selection.

**Comment by bksubhuti** [comment]
If we use `<p class"body text" id="M1.0005">yopi </p> `, then the html will be selectable.
however, it is part of our strategy to have this for displaying alternate page numbering when turned on (there are different printed versions of the pali text).  Is it possible not to use an empty sized box as you say is being done?

Or is this done more upstream as the issue I posted.

**Comment by bksubhuti** [comment]
Should we complain at this link below?  
https://github.com/flutter/flutter/issues/111021

Or is it something that can be fixed in html widget?
I think if you do nothing instead of SizedBox.shrink then this would be much better for us.

**Comment by daohoangson** [comment]
I have been able to reproduce it reliably with this sample

https://dartpad.dev/?id=76db6a5a5629b306b0d00fcf8c773f81

```dart
    Text.rich(
      TextSpan(
        children: [
          TextSpan(text: 'Lorem'),
          WidgetSpan(child: SizedBox(width: 4)),
          TextSpan(
            text: 'ipsum dolor sit amet, consectetur adipiscing elit. ',
          ),
        ],
      ),
    );
```

It appears text selection will fail if the paragraph has more than 1 line and there is a widget span in the middle. I have opened flutter/flutter#130427 to report this.

**Comment by daohoangson** [comment]
A contributor confirmed the bug and said it has been fixed in Flutter master. I tried your HTML with the latest build and it did work as expected:


<img width="827" alt="Screen Shot 2023-07-13 at 14 17 31" src="https://github.com/daohoangson/flutter_widget_from_html/assets/239336/f5643a78-e286-4cc2-8a32-43f2d59353c0">

**Comment by bksubhuti** [comment]
Your research, help and kindness has been very much appreciated.  
Our noncommercial app is up to 2400 active installs on android and maybe 
another 1000 on desktop and apple.  I will confirm the flutter master build later today and then close the issue.

**Comment by bksubhuti** [comment]
We have a few upgrade problems with "master" due to some neglected depreciated clipboard calls, but I was able compile and test the code under master. 
This issue is closed.  

We appreciate your work.  Because of this package, we were able to move forward with 5 OS builds which was the game changer and a major goal for the project since our core reading data is html based.

The gesture detector suggested earlier also worked and helped solve more than one problem.
